File: Online_Judge_Problems/boj/boj_20949_new_monitor/boj_20949_v2.js
/**
 * -----------------------------------------------------------
 * Sub    : [BOJ] 효령과 새 모니터
 * Link   : https://www.acmicpc.net/problem/20949
 * Level  : Silver 5
 * Tag    : JS, sort
 * ------------------------------------------------------------
 * Details
 *
 * ------------------------------------------------------------
 */

let input;

if (process.platform === "linux") {
  input = require("fs").readFileSync(0, "utf-8").trim().split("\n");
} else {
  const path = require("path");
  const filePath = path.join(__dirname, "input.txt");
  input = require("fs").readFileSync(filePath, "utf-8").trim().split("\n");
}

// input
const N = parseInt(input[0]);
const monitors = [];

for (let i = 1; i <= N; i++) {
  const [W, H] = input[i].split(" ").map(Number);

  // (일반 Number 타입으로도 이 범위는 커버 가능하지만 정확성을 위해 권장)
  const ppiValue = BigInt(W) * BigInt(W) + BigInt(H) * BigInt(H);

  monitors.push({
    id: i,
    ppiValue: ppiValue,
  });
}

// sort
monitors.sort((a, b) => {
  if (b.ppiValue > a.ppiValue) return 1;
  if (b.ppiValue < a.ppiValue) return -1;

  // PPI가 같으면 번호(id)가 작은 순서대로
  return a.id - b.id;
});

// print
console.log(monitors.map((m) => m.id).join("\n"));

File: Online_Judge_Problems/boj/boj_20949_new_monitor/boj_20949_v2.py
"""
-----------------------------------------------------------
Sub    : [BOJ] 효령과 새 모니터
Link   : https://www.acmicpc.net/problem/20949
Level  : 실버 5
Tag    : Python, sort
-----------------------------------------------------------
Details

-----------------------------------------------------------
"""

import os
import sys

if sys.platform == "linux":
    lines = sys.stdin.read().splitlines()
else:
    file_path = os.path.join(os.path.dirname(__file__), "input.txt")
    if os.path.exists(file_path):
        with open(file_path, "r") as f:
            lines = f.read().splitlines()
    else:
        lines = []

input = iter(lines).__next__


def solve():
    try:
        N = int(input())
        monitors = []
        for i in range(1, N + 1):
            W, H = map(int, input().split())
            monitors.append((W**2 + H**2, i))

        # 정렬 및 출력 로직...
        monitors.sort(key=lambda x: (-x[0], x[1]))
        print("\n".join(str(m[1]) for m in monitors))

    except (StopIteration, ValueError):
        # 데이터가 없거나 입력이 끝났을 때 에러 방지
        pass


if __name__ == "__main__":
    solve()
